### PR TITLE
feat(java): Add automatic and custom performance metrics doc

### PR DESCRIPTION
**Pull request by markushi**
Add missing API doc for custom performance metrics, including java and kotlin example code.

See https://github.com/getsentry/sentry-java/pull/2260 for the relevant SDK implementation.

@romtsn who else should be reviewing this PR?

### CODE REVIEW

**Comment by romtsn** [comment]
@markushi usually you can add the entire mobile team to reviewers and whoever approves first wins :)

**Review by krystofwoldrich — APPROVED** [review]

**Comment by romtsn** [comment]
LGTM 👍 but we need to release the SDK 6.5.0 first, so I'd hold this off until it's released. Maybe convert it into a draft to not merge accidentally

**Review by romtsn — APPROVED** [review]

**Comment by markushi** [comment]
- [ ] Wait for sentry-java 6.5.0 release

**Comment by AbhiPrasad** [comment]
While we are here, could we add a page that describes our automatic performance metrics for Android, under here: https://github.com/getsentry/sentry-docs/tree/master/src/platform-includes/performance/automatic-performance-metrics (you can follow the same format as JavaScript)

File: src/platform-includes/performance/automatic-performance-metrics/android.mdx [hunk 1]
```diff
@@ -0,0 +1,4 @@
+If configured, the Android SDK can automatically collect the following performance metrics:
+
+* Cold and Warm app start time, see [App Start instrumentation](/platforms/android/performance/instrumentation/automatic-instrumentation/#app-start-instrumentation).
+* Frozen and slow frame rendering, details can be found on the [Slow and Frozen Frames page](/platforms/android/performance/instrumentation/automatic-instrumentation/#slow-and-frozen-frames).
```
Comment (imatwawana) [inline src/platform-includes/performance/automatic-performance-metrics/android.mdx:4]:
```suggestion
* Cold and Warm app start time - learn more in [App Start Instrumentation](/platforms/android/performance/instrumentation/automatic-instrumentation/#app-start-instrumentation).
* Frozen and slow frame rendering - learn more in [Slow and Frozen Frames](/platforms/android/performance/instrumentation/automatic-instrumentation/#slow-and-frozen-frames).
```

**Review by imatwawana — COMMENTED** [review]
Made some wording and style edits and left a question

File: src/platform-includes/performance/automatic-performance-metrics/android.mdx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+If configured, the Android SDK can automatically collect the following performance metrics:
```
[the remark this answers is not in the document]
Comment (markushi) [reply src/platform-includes/performance/automatic-performance-metrics/android.mdx:1]: Thanks for your suggestion! Yes, the SDK automatically captures those metrics by default. There are options to disable this but it's on by default.

File: src/platform-includes/performance/automatic-performance-metrics/android.mdx [hunk 3]
```diff
@@ -0,0 +1,4 @@
+If configured, the Android SDK automatically collects the following performance metrics:
+
+* Cold and Warm app start time - learn more in [App Start Instrumentation](/platforms/android/performance/instrumentation/automatic-instrumentation/#app-start-instrumentation).
```
Comment (philipphofmann) [inline src/platform-includes/performance/automatic-performance-metrics/android.mdx:3]: `l`:
```suggestion
* Cold and warm app start time - learn more in [App Start Instrumentation](/platforms/android/performance/instrumentation/automatic-instrumentation/#app-start-instrumentation).
```

File: src/platforms/common/performance/instrumentation/performance-metrics.mdx [hunk 4]
```diff
@@ -31,7 +31,7 @@ description: "Learn how to attach performance metrics to your transactions."
 
 Sentry's SDKs support sending performance metrics data to Sentry. These are numeric values attached to transactions that are aggregated and displayed in Sentry.
 
-<PlatformSection supported={["javascript"]} notSupported={["react-native"]}>
+<PlatformSection supported={["javascript", "android"]} notSupported={["react-native"]}>
```
Comment (philipphofmann) [inline src/platforms/common/performance/instrumentation/performance-metrics.mdx:34]: `m`: This should also work on Java, I guess.
```suggestion
<PlatformSection supported={["javascript", "java", "android"]} notSupported={["react-native"]}>
```

**Review by philipphofmann — APPROVED** [review]
Two minor comments, thanks, @markushi.

File: src/platforms/common/performance/instrumentation/performance-metrics.mdx (hunk 4 above)
Comment (markushi) [reply src/platforms/common/performance/instrumentation/performance-metrics.mdx:34]: This section includes automatic performance metrics - in this case I think it's only relevant on Android (slow/cold start, slow/frozen frames).

Custom metrics are mentioned down below and are available for both Java and Android.

**Pull request merged**